### ISSUE TITLE
BUG?  Draw feature

### ISSUE THREAD

**Issue by everythingability** [benjie-git/CardStock#18]
When I use the pencil, I would expect (perhaps) to be able to define/drag an area to create a graphic canvas and then to use the pencil in it... but each time my mouseups, it's creating a new pencil line, making "drawing" something that isn't easy... single dots are impossible.

Perhaps both object types might be creatable... a vector points based pen? and more of a Mac Paint like means of drawing... pixels etc.

**Comment by benjie-git** [comment]
I agree that this would be great.  I'd love to add a full paint canvas object type, complete with the spray can!  But it hasn't make it to the top of my priority list yet.  But I certainly agree that the pencil is a bit odd as it stands.  Maybe I could start by changing the pencil to be an object type that creates a rectangular canvas that you can then keep pencil-drawing into until you double click, or press enter, or click the hand/select tool.  And then add other paint tools later.

**Comment by everythingability** [comment]
Yeah. My half arsed vision would be

a. Click the draw defining area tool (a pencil)
b. Cursor changes to a cross 
c. I draw a rect
d. When mouse is in rect cursor changes to a pencil/brush/spray/rubber/fill etc.
e. When I click outside, maybe selected tool reverts to pointer? Or does it only change when a different tool is actually clicked? 

As I play around, other muscle memory items are missing. I'd really like command-M for the message window.... and be able to switch run/edit with command-tab etc.

**Comment by benjie-git** [comment]
I like the idea of including a MacPaint-in-a-canvas object.

I'm not planning to take HyperCard keyboard shortcut muscle memory into account, due to a lapsed statute of limitations on that one.  :)
